encodings/README.md

```diff
@@ -0,0 +1,149 @@
+# OCSF Encodings
+
+This repository contains information, recommendations, and tools to
+support representations of events described by the OCSF abstract
+schema.  Representations include both *encodings*, which are ways of
+serializing OCSF events into sequences of bytes, and *in memory*
+representations, which support managing OCSF events to native objects
+within particular languages or systems.  (Some encodings, like
+**protobuf** blur the distinction because they have a well defined
+serialization but also their own ecosystem for creating language
+representations based on the abstract protobuf description.)
+
+## Organization of this Repository
+
+This repository is organized by _representation_, so for example
+information and tools to support JSON encodings are found in the
+[`json/`](json/README.md) directory.
+
+## Process for Developing an Encoding
+
+This describes the general process to develop a new encoding.
+
+### Use Cases
+
+Identify and articulate the use cases that the encoding is intended to
+target.  This will not only help guide choices in the encoding mapping
+process, but also inform others as to when a particular encoding might
+be useful to adopt.
+
+### Primitive Data Types
+
+Map the OCSF primitive data types to those available in the encoding.
+
+The OCSF primitive types are the types in the dictionary
+(`dictionary.json`, `types`) which do *not* have a `type` property.
+This is the current list of all primitive types:
+
+| Type        | Notes                            |
+| ----------- | -------------------------------- |
+| `boolean_t` | `true` or `false`                |
+| `float_t`   | possibly non-integer number      |
+| `integer_t` | smallish (~32 bit) integer       |
```

Comment:
It seems you've generally taken the approach of writing down the current state, which is good.

On `integer_t` it seems the current state is just that it is a "signed integer". I haven't seen anything commenting on the range of values it should be used for.

Reply:
I put "smallish" because there is also a `long_t` type which is there to represent "largish" integers.  The constraints live in the schema, when they are present.  In casual conversation, the implication seems to have been that `integer_t` is for 32 bit ints and `long_t` is for 64 bit ints.  But, to your point, that has not been crystal clear.

My preference would be to abstract that out at the schema level and get rid of `long_t` entirely and require range constraints on every usage of  `integer_t`.

Reply:
that said, I added some extra clarification.  Still a WIP of course!

Reply:
Why not define `int_t` as "signed 32-bit integer" and `long_t` as "signed 64-bit integer"? The `long_t` type is already defined as a "signed 8-byte integer", which means the same thing. The actual type of `int_t` isn't defined, but given the history of OCSF, it seems clear that it is a 32-bit value.

Reply:
Agree on the history, but I am very much leaning _away_ from specifying that at the primitive data type level.

The primitive types, in my view, should be abstract.  An integer is a mathematical construct, and the abstract schema should enjoy that.  _But_ the dictionary can and should (future state IMHO: must) define constraints that limit the range and precision of numbers.

I'd even go so far as saying there's just a `number` abstract type, and an integer is simply a number with 0 digits to the right of the decimal point.  That might be spicier than we want to go though.

Reply:
Concrete types are, I think, necessary for encodings, even with JSON. Wouldn't abstract type be problematic? (Perhaps I'm missing something.)

Reply:
(All that said, this debate is no reason to hold up this PR.)